src/dev/src/app/components/tree-view/tree-view-client-side-demo.ts

```diff
@@ -0,0 +1,480 @@
+          return { data: it, label: 'name' };
+        });
+      );
+      }
+    }
+    
+    treeViewConfiguration: TreeViewConfiguration = {
+      width: '250px',
+      height: '300px'
+    }
+   `;
+
+  dataExample2 = `
+    // Developer can pass optional parameters for TreeView configuration
+    // Otherwise default ones are issued
+    treeViewConfiguration: TreeViewConfiguration = {
+      width: '250px',
+      height: '300px',
+      hasBorders: true,
+      throttleTime: 250,
+      bufferAmount: 500
+    }
+    serverDataRetriever: PagedTreeNodeDataRetriever<FoodNode> = {
+    hasChildNodes: (node: TreeNode<FoodNode>) => {
+      return Promise.resolve(node.data.hasChildren());
+    },
+    // not used on Server Side rendering
+    getChildNodeData: (node: TreeNode<FoodNode>) => {
+      return Promise.resolve();
+    },
+    getPagedChildNodeData: (node: TreeNode<FoodNode>, pagingParams: PagingParams) => {
+      const url = 'https://back-end-server/' + node.data.id + '?start=' +
+                  pagingParams.offset + '&limit=' + pagingParams.limit;
+      return this.http.get(url).subscribe((results: TreeNodeData[]) => {
+        return results;
+      });
+     },
+    };
+  `;
+  dataExample3 = `
+    // When using non root tree view, pass in an empty instance of the NonRootTreeNode class
+    const noRootTreeView = new NonRootTreeNode();
+    
+    // Developer can add the getIconTemplate & getNodeTemplate methods to TreeNodeDataRetriever
+    // to customize node how they wish
+    treeDataArrayRetriever: TreeNodeDataRetriever<PersonInterface> = {
+      
+      hasChildNodes: (node: TreeNode<PersonInterface>) => {
+        return Promise.resolve(!!node.data.data.children && node.data.data.children.length > 0);
+      },
+      
+      getChildNodeData: (node: TreeNode<PersonInterface>) => {
+        const isEmptyRoot = node.data instanceof NonRootTreeNode;
+        if (isEmptyRoot) {
+          return Promise.resolve(
+            this.jsonData.map(it => {
+              return { data: it, label: 'name' };
+            });
+          );
+        }
+        return Promise.resolve(
+          node.data.data.children.map(it => {
+            return { data: it, label: 'name' };
+          });
+        );
+      },
+      
+      // Icon templates will be used as the clickable icons to expand or collapse node
+      getIconTemplate: (node: TreeNode<PersonInterface>): TemplateRef<any> => {
+        return node.expanded ? this.expandedTem : this.nonExpandedTem;
+      },
+      
+      // Node template must take in a node property to adjust label according to developer wants
+      getNodeTemplate: (): TemplateRef<any> => {
+        return this.nodeTemplate;
+      },
+    };
+  `;
+
+  treeNodeData: TreeNodeData<FoodNode> = {
+    data: FOOD_TREE_DATA,
+    label: 'name',
+  };
+  serverSideTreeNodeData: TreeNodeData<FoodNode> = {
+    data: { name: 'Foods' },
+    label: 'name',
+  };
+  noRoot = new NonRootTreeNode();
```

Comment:
This noRoot instance is for us to know to not display the treeNodeData until we get the first array of tree nodes from the developer. The initial Input of the tree view is a treeNodeData not an array of so this may we know to start display at the second level